docs/index.md

```diff
@@ -158,8 +158,41 @@ When an issuer generates a new key to sign Health Cards, the public key SHALL be
 issuer's JWK set in its `jwks.json` file. Retired private keys that are no longer used to sign Health Cards SHALL be destroyed.
 Older public key entries that are needed to validate previously
 signed Health Cards SHALL remain in the JWK set for as long as the corresponding Health Cards
-are clinically relevant. However, if a private signing key is compromised, then the issuer SHALL immediately remove the corresponding public key
-from the JWK set in its `jwks.json` file and request revocation of all X.509 certificates bound to that public key.
+are clinically relevant. However, if a private signing key is compromised, then the issuer SHALL immediately remove the corresponding public key from the JWK set in its `jwks.json` file and request revocation of all X.509 certificates bound to that public key; verifiers will from then on reject all Health Cards signed using that key.
+
+### Revocation
+
+Individual Health Cards MAY be revoked by an issuer using the digest of their JWS representation. The Revocation Digest Fragment (RDF) of a Health Card is calculated as follows:
+1. Calculate the SHA-256 `<<digest>>` of the utf-8 encoding of the `<<Health Card as JWS>>`,
```

Comment:
An alternative to this would be for the issuer to compute the digest (or its unique prefix) of the JWS **excluding** the signature and to combine that digest with a revocation nbf value, i.e., the revocation identifier would be `<unique prefix of signatureless digest>-<revocation nbf>`. This would revoke, for the given kid, any cards with the given signatureless digest value and also containing an nbf value less than or equal to the revocation nbf.  

As the signature has a random component, issuers are not able to revoke a card using the currently proposed identification scheme unless they keep track of sufficient information for each card issued or if they can recover that information after the fact (e.g., the card is retained by a border agent that suspects fraud).  

This alternative scheme would allow issuers to revoke cards without regard for the card's signature and would also allow a card with the exact same information to be reissued with the same signing key by setting the card's nbf value higher than that of the revocation nbf.

Reply:
For our current profiles, this would leave no guaranteed source of entropy in the inputs for the hash digest. This means it would be feasible to probe (name, birthday, vaccine type, vaccine date) tuples and discover source information for revoked SHCs (especially if you start with a population-level list of actual names and birthdates, which is available from large scale data breaches).

Reply:
I'm not sure how feasible it is for someone to correlate large data breaches that may contain names and birthdays with vaccine date and type to perform queries on CRLs.  It is more likely that a fraudster would attempt to query the CRL for their own (or sold) fraudulent SHCs.  But in that scenario, the fraudster would have access to all the non-deterministic data too, and thus able to perform the query in either case. 

I'm more concerned about the complications introduced in this proposal.  In the first variant, the verifier must strip the nbf out of the JSON correctly before computing the digest.  While the second variant avoids this complication with JSON serialization, it relies upon the issuer keeping track of nbs values which, to me, is no different than keeping track of RDFs.  Furthermore, it was mentioned in D204 that the choice to go with a counter versus timestamp in the CRL was due issuer mistakes seen with nbf values, but this proposal relies upon those.

Reply:
The idea of having a time interval for invalid usage of a kid is interesting. Could be per SHC before a `nbf` as suggested above, but could be useful for all SHC for a specific time period. E.g., if a large issuer (a state/province) knows that invalid cards have been issued by a rogue insider during a specific period (e.g. one week), then advertising an invalid period would be better than revoking the key itself (therefore, invalidating all issued cards).